From e77f4d199040604b022f463cb89f42650d40bb54 Mon Sep 17 00:00:00 2001
From: Rashed Abdel-Tawab <rashed@linux.com>
Date: Wed, 23 Nov 2016 23:00:37 -0500
Subject: [PATCH 20/39] extract_utils: Use shasum on macOS

macOS doesn't have sha1sum but instead shasum, so use that
if 'uname -s' returns Darwin

Change-Id: I1ab8721327a561e8df78562bbc0b12e884cc4a93
---
 build/tools/extract_utils.sh | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/build/tools/extract_utils.sh b/build/tools/extract_utils.sh
index 6e8baa3c..12537006 100644
--- a/build/tools/extract_utils.sh
+++ b/build/tools/extract_utils.sh
@@ -858,13 +858,21 @@ function extract() {
                 if [ ! -f "$DEST" ]; then
                     KEEP="1"
                 else
-                    local DEST_HASH=$(sha1sum "$DEST" | awk '{print $1}' )
+                    if [ "$(uname)" == "Darwin" ]; then
+                        local DEST_HASH=$(shasum "$DEST" | awk '{print $1}' )
+                    else
+                        local DEST_HASH=$(sha1sum "$DEST" | awk '{print $1}' )
+                    fi
                     if [ "$DEST_HASH" != "$HASH" ]; then
                         KEEP="1"
                     fi
                 fi
                 if [ "$KEEP" = "1" ]; then
-                    local TMP_HASH=$(sha1sum "$TMP" | awk '{print $1}' )
+                    if [ "$(uname)" == "Darwin" ]; then
+                        local TMP_HASH=$(shasum "$TMP" | awk '{print $1}' )
+                    else
+                        local TMP_HASH=$(sha1sum "$TMP" | awk '{print $1}' )
+                    fi
                     if [ "$TMP_HASH" = "$HASH" ]; then
                         printf '    + (keeping pinned file with hash %s)\n' "$HASH"
                         cp -p "$TMP" "$DEST"
-- 
2.15.0

